subprojects/launcher/src/main/java/org/gradle/launcher/daemon/server/health/DaemonHealthStats.java

```diff
@@ -32,20 +34,25 @@
 public class DaemonHealthStats {
 
     private final DaemonRunningStats runningStats;
-    private final MemoryInfo memoryStatus;
     private final GarbageCollectionInfo gcInfo;
     private final GarbageCollectionMonitor gcMonitor;
+    private JvmMemoryStatus memoryStatus;
 
-    public DaemonHealthStats(DaemonRunningStats runningStats, MemoryInfo memoryInfo, ScheduledExecutorService scheduledExecutorService) {
-        this(runningStats, memoryInfo, new GarbageCollectionInfo(), new GarbageCollectionMonitor(scheduledExecutorService));
+    public DaemonHealthStats(DaemonRunningStats runningStats, MemoryManager memoryManager, ScheduledExecutorService scheduledExecutorService) {
+        this(runningStats, memoryManager, new GarbageCollectionInfo(), new GarbageCollectionMonitor(scheduledExecutorService));
     }
 
     @VisibleForTesting
-    DaemonHealthStats(DaemonRunningStats runningStats, MemoryInfo memoryStatus, GarbageCollectionInfo gcInfo, GarbageCollectionMonitor gcMonitor) {
+    DaemonHealthStats(DaemonRunningStats runningStats, MemoryManager memoryManager, GarbageCollectionInfo gcInfo, GarbageCollectionMonitor gcMonitor) {
         this.runningStats = runningStats;
-        this.memoryStatus = memoryStatus;
         this.gcInfo = gcInfo;
         this.gcMonitor = gcMonitor;
+        memoryManager.addListener(new JvmMemoryStatusListener() {
+            @Override
+            public void onJvmMemoryStatus(JvmMemoryStatus jvmMemoryStatus) {
+                memoryStatus = jvmMemoryStatus;
```

Comment:
Possibly need some kind of synchronisation here. The listener is called on a different thread to that using the memory status.